client/src/components/cv/ContactsListCV.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { List, Typography } from 'antd';
+import {
+  PhoneOutlined,
+  MailOutlined,
+  SkypeOutlined,
+  LinkedinOutlined,
+  AimOutlined,
+  GithubOutlined,
+  IdcardOutlined,
+} from '@ant-design/icons';
+import { Contacts, ContactType } from '../../../../common/models/cv';
+
+const { Item } = List;
+const { Text } = Typography;
+
+type AllowedContacts = {
+  [key in ContactType]: {
+    icon: React.ReactNode;
+    transformFunc?: (contact: string) => React.ReactNode;
+  };
+};
+
+const allowedContacts: AllowedContacts = {
+  phone: {
+    icon: <PhoneOutlined />,
+  },
+  email: {
+    icon: <MailOutlined />,
+    transformFunc: contact => (
+      <a className="black-on-print" href={`mailto:${contact}`}>
+        {contact}
+      </a>
+    ),
+  },
+  skype: {
+    icon: <SkypeOutlined />,
+    transformFunc: contact => (
+      <a className="black-on-print" href={`skype:${contact}?chat`}>
+        {contact}
+      </a>
+    ),
+  },
+  telegram: {
+    icon: <PhoneOutlined />,
+    transformFunc: contact => (
+      <a className="black-on-print" href={`https://t.me/${contact}`}>
+        {contact}
+      </a>
+    ),
+  },
+  linkedin: {
+    icon: <LinkedinOutlined />,
+    transformFunc: contact => (
+      <a className="black-on-print" href={`https://www.linkedin.com/in/${contact}`}>
+        {contact}
+      </a>
+    ),
+  },
+  location: {
+    icon: <AimOutlined />,
+  },
+  github: {
+    icon: <GithubOutlined />,
+    transformFunc: contact => (
+      <a className="black-on-print" href={`https://github.com/${contact}`}>
+        {contact}
+      </a>
+    ),
+  },
+  website: {
+    icon: <IdcardOutlined />,
+    transformFunc: contact => (
+      <a className="black-on-print" href={contact}>
+        {contact}
+      </a>
+    ),
+  },
+};
+
+type Props = {
+  contacts: Contacts | null;
+};
+
+type EntryOf<T extends object> = { [K in keyof T]: [K, T[K]] }[keyof T];
+
+function ContactsListCV(props: Props) {
+  const { contacts } = props;
+  let res;
+  if (contacts !== null) {
+    const contactsFiltered = Object.entries(contacts);
+    const renderContacts = contactsFiltered.filter((contact): contact is EntryOf<Contacts> => contact[1] !== null);
```

Comment:
Why do you name it like function?
By it name I think that it's some function that render contacts.

Reply:
You can rewrite it like this:
```
const omitEmpty = ([, contactText]): contact is EntryOf<Contacts> => contactText !== null);
const contactsFiltered = Object.entries(contacts).filter(omitEmpty);
```